debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/ParallelSnapshotReaderTest.java

```diff
@@ -159,69 +159,75 @@ public void testHaltingPredicateHonorsTimeRange() {
+        Duration duration = Duration.ofSeconds(5 * 60); // five minutes
 
-        boolean testResult = parallelHaltingPredicate.test(createSourceRecordWithTimestamp(System.currentTimeMillis()));
+        ParallelHaltingPredicate parallelHaltingPredicate = new ParallelHaltingPredicate(thisReaderNearEnd, otherReaderNearEnd, duration);
 
-        Assert.assertFalse(testResult);
+        boolean testResult = parallelHaltingPredicate.test(createSourceRecordWithTimestamp(System.currentTimeMillis()/1000));
+
+        Assert.assertTrue(testResult);
 
-        Assert.assertTrue(thisReaderBoolean.get());
-        Assert.assertFalse(otherReaderBoolean.get());
+        Assert.assertTrue(thisReaderNearEnd.get());
+        Assert.assertFalse(otherReaderNearEnd.get());
     }
 
     @Test
     public void testHaltingPredicateHalts() {
-        // verify that the halting predicate returns true if both the `this` and
-        // `other` reader booleans are true.
-
-        AtomicBoolean thisReaderBoolean = new AtomicBoolean(false);
-        AtomicBoolean otherReaderBoolean = new AtomicBoolean(true);
+        // verify that the halting predicate returns false if both the 'this' and
+        // 'other' reader are near the end of the binlog.
 
-        Duration duration = Duration.ofMillis(5 * 60 * 1000); // five minutes
+        AtomicBoolean thisReaderNearEnd = new AtomicBoolean(false);
+        AtomicBoolean otherReaderNearEnd = new AtomicBoolean(true);
 
-        ParallelHaltingPredicate parallelHaltingPredicate = new ParallelHaltingPredicate(thisReaderBoolean, otherReaderBoolean, duration);
+        Duration duration = Duration.ofSeconds(5 * 60); // five minutes
 
-        boolean testResult = parallelHaltingPredicate.test(createSourceRecordWithTimestamp(System.currentTimeMillis()));
+        ParallelHaltingPredicate parallelHaltingPredicate =
+            new ParallelHaltingPredicate(thisReaderNearEnd, otherReaderNearEnd, duration);
 
-        Assert.assertTrue(testResult);
+        boolean testResult =
+            parallelHaltingPredicate.test(createSourceRecordWithTimestamp(System.currentTimeMillis()/1000));
 
-        Assert.assertTrue(thisReaderBoolean.get());
-        Assert.assertTrue(otherReaderBoolean.get());
+        Assert.assertFalse(testResult);
+        
+        Assert.assertTrue(thisReaderNearEnd.get());
+        Assert.assertTrue(otherReaderNearEnd.get());
     }
 
     /**
      * Create an "offset" containing a single timestamp element with the given value.
-     * @param tsMs the timestamp in the resulting offset.
+     * @param tsSec the timestamp (in seconds) in the resulting offset.
      * @return an "offset" containing the given timestamp.
      */
-    private SourceRecord createSourceRecordWithTimestamp(long tsMs) {
-        Map<String, ?> offset = Collections.singletonMap(SourceInfo.TIMESTAMP_KEY, tsMs);
+    private SourceRecord createSourceRecordWithTimestamp(long tsSec) {
```

Comment:
If this method gets changed anyways, let's rather use `Instant` or any other suitable Java 8 temporal type (it's something we generally strive for when adding/touching code handling time info).

Reply:
That seems like a little heavyweight of a solution just for a test mock record.